lib/peep_message.rb

```diff
@@ -0,0 +1,46 @@
+require_relative './database_connection'
+
+class PeepMessage
+
+  attr_reader :id, :message, :name, :username, :time
+
+  def initialize(id, message, name, username, time)
+    @id = id
+    @message = message
+    @name = name
+    @username = username
+    @time = time
+  end
+
+  def ==(other)
+    @id == other.id
+  end
+
+  def self.all
+    result = DatabaseConnection.query("SELECT * FROM peeps")
+    result.to_a.reverse.map do |peep|
+      PeepMessage.new(peep['id'], peep['message'], peep['name'],
+                      peep['username'], peep['time'])
+    end
+  end
+
+  def self.create(message, name, username, time)
```

Comment:
What was the thinking behind splitting your create and insert_into_database methods into two.  I don't think it's wrong as such, but conventionally I would expect the create method to be responsible for writing to the database.  By using the SQL RETURNING keyword (as you do on line 43), you can achieve the same result of returning a Ruby object.

Some thing like:

```ruby
def self.create(message, name, username, time)
  result = DatabaseConnection.query(
+      "INSERT INTO peeps (message, name, username, time)
+      VALUES ('#{message}', '#{name}', '#{username}', '#{time}')
+      RETURNING peep_id, message, name, username, time"
+ )
   PeepMessage.new(result.first['id'], peep_message = result.first['message'], peep_name = result.first['name'], peep_username = result.first['username'], peep_time = result.first['time'])
end
```